pkg/services/sqlstore/migrations/db_file_storage.go

```diff
@@ -2,37 +2,46 @@ package migrations
 
 import "github.com/grafana/grafana/pkg/services/sqlstore/migrator"
 
-// TODO: remove nolint as part of https://github.com/grafana/grafana/issues/45498
-// nolint:unused,deadcode
 func addDbFileStorageMigration(mg *migrator.Migrator) {
 	filesTable := migrator.Table{
 		Name: "file",
 		Columns: []*migrator.Column{
 			{Name: "path", Type: migrator.DB_NVarchar, Length: 1024, Nullable: false},
-			{Name: "parent_folder_path", Type: migrator.DB_NVarchar, Length: 1024, Nullable: false},
+			{Name: "path_hash", Type: migrator.DB_NVarchar, Length: 64, Nullable: false},
```

Comment:
Table structures of `file` and `file_metadata` are the key things in this PR, I plan to improve/document the implementation in https://github.com/grafana/grafana/issues/48239 cc @zserge